src/core/component.js

```diff
@@ -268,6 +269,8 @@ Component.prototype = {
       if (el.initializingComponents[this.name]) { return; }
       // Prevent infinite loop in case of init method setting same component on the entity.
       el.initializingComponents[this.name] = true;
+      // Execute decorator functions on component before init
+      executeDecorators(this);
```

Comment:
Is this avoidable, can we do pass in `this` explicitly?

Reply:
Unfortunately that's the trick to get this all working. I couldn't find any way of avoiding it.. although I thought a lot about it. In my own node package I had to monkey patch this call in by wrapping some core functions